chapter2/tests/t12/nginx.conf

```diff
@@ -0,0 +1,73 @@
+user nginx nginx;
+worker_processes 1;
+error_log /tmp/camp/error.log notice;
+access_log /tmp/camp/access.log main;
+pid /run/nginx.pid;
+
+events {
+	worker_connections 1024;
+}
+
+http {
+
+	log_format main '$remote_addr - $remote_user [$time_local] "$request" '
+	'$status $body_bytes_sent "$http_referer" '
+	'"$http_user_agent" "$http_x_forwarded_for"';
+
+	keepalive_timeout	65s;
+	types_hash_max_size	4096;
+	include /etc/nginx/mime.types;
+	default_type application/octet-stream;
+	index index.html;
+
+	upstream python-uwsgi {
+		server 127.0.0.1:5000;
+	}
+
+	server {
+		listen 443 ssl http2;
+		listen [::]:443 ssl http2;
+		server_name _;
+		root /tmp/nginx;
+		ssl_certificate "/tmp/nginx/certs/cert.pem";
+		ssl_certificate_key "/tmp/nginx/certs/key.pem";
+		ssl_session_cache shared:SSL:1m;
+		ssl_session_timeout 10m;
+		ssl_ciphers PROFILE=SYSTEM;
+		ssl_protocols TLSv1.2 TLSv1.3;
+		ssl_ciphers ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:\
+		ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:\
+		ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384;
+		ssl_prefer_server_ciphers on;
+		error_page 404 /404.html;
+		location = /404.html {
+		}
```

Comment:
prefer to have space before any location/server/upstream contexts - as they're important

Reply:
> prefer to have space before any location/server/upstream contexts - as they're important

@dmitry-mightydevops 
Done